core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -292,21 +283,32 @@ private void write0(HttpObject o, boolean endOfStream, boolean flush) {
             throw new Error();
         }
 
-        if (endOfStream) {
-            logBuilder().endResponse();
-        }
+        future.addListener((ChannelFuture f) -> {
+            if (f.isSuccess()) {
+                // Write an access log only if the last message has been sent.
```

Comment:
What if we sent a partial response (or processed the request but failed to send any response)? Shouldn't we log somehow if that was the case?

Reply:
For the case you mentioned, write an access log [here](https://github.com/line/armeria/pull/895/files#diff-b13dd0316631db6a58eb2a29b0f67f5fR302). I will fix the comment. :-)

Reply:
Aha. So, there are two places the perform logging. OK!